batch/batch/server/server.py

```diff
@@ -1111,10 +1088,22 @@ def __anext__(self):
         await asyncio.sleep(REFRESH_INTERVAL_IN_SECONDS)
 
 
+async def db_cleanup_event_loop():
+    await asyncio.sleep(60)
+    while True:
+        try:
+            batches = [Batch.from_record(record) for record in await db.batch.get_finished_deleted_records()]
+            await asyncio.gather(*[batch.delete() for batch in batches])
```

Comment:
This is a  low priority cleanup thread.  No need to run these delete operations concurrently.  Just loop over the records and call delete on the batches one by one.